website/docs/r/role_assignment_marketplace.html.markdown

```diff
@@ -0,0 +1,113 @@
+
+```hcl
+data "azurerm_client_config" "example" {
+}
+
+data "azurerm_role_definition" "example" {
+  name = "Log Analytics Reader"
+}
+
+resource "azurerm_role_assignment_marketplace" "example" {
+  role_definition_id = "${data.azurerm_role_definition.example.id}"
+  principal_id       = data.azurerm_client_config.example.object_id
+
+  lifecycle {
+    ignore_changes = [
+      role_definition_name,
+    ]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `principal_id` - (Required) The ID of the Principal (User, Group or Service Principal) to assign the Role Definition to. Changing this forces a new resource to be created.
+
+~> **NOTE:** The Principal ID is also known as the Object ID (i.e. not the "Application ID" for applications). To assign Azure roles, the Principal must have `Microsoft.Authorization/roleAssignments/write` permissions. See [documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal) for more information.
+
+* `name` - (Optional) A unique UUID/GUID for this Role Assignment - one will be generated if not specified. Changing this forces a new resource to be created.
+
+* `role_definition_id` - (Optional) The Scoped-ID of the Role Definition. Changing this forces a new resource to be created. Conflicts with `role_definition_name`.
+
+* `role_definition_name` - (Optional) The name of a built-in Role. Changing this forces a new resource to be created. Conflicts with `role_definition_id`.
+
+~> **NOTE:** To assign `Marketplace Admin` role, the calling Principal must first be assigned Privileged Role Administrator (like `Owner` role) or Global Administrator. See [documentation](https://learn.microsoft.com/en-us/marketplace/create-manage-private-azure-marketplace-new#prerequisites) for more information.
+
+* `condition` - (Optional) The condition that limits the resources that the role can be assigned to. Changing this forces a new resource to be created.
+
+* `condition_version` - (Optional) The version of the condition. Possible values are `1.0` or `2.0`. Changing this forces a new resource to be created.
+
+* `delegated_managed_identity_resource_id` - (Optional) The delegated Azure Resource ID which contains a Managed Identity. Changing this forces a new resource to be created.
+
+~> **NOTE:** this field is only used in cross tenant scenario.
+
+* `description` - (Optional) The description for this Role Assignment. Changing this forces a new resource to be created.
+
+* `skip_service_principal_aad_check` - (Optional) If the `principal_id` is a newly provisioned `Service Principal` set this value to `true` to skip the `Azure Active Directory` check which may fail due to replication lag. This argument is only valid if the `principal_id` is a `Service Principal` identity. Defaults to `false`. Changing this forces a new resource to be created.
+
+~> **NOTE:** If it is not a `Service Principal` identity it will cause the role assignment to fail.
```

Comment:
This note isn't clear on what we're expecting from the user - is this note saying that the `principal_id` cannot be set to any other type other than a Service Principal? Or should `skip_service_principal_aad_check` always be set to `false` if `principal_id` is something other than a Service Principal?
```suggestion
~> **NOTE:** If it is not a `Service Principal` identity it will cause the role assignment to fail.
```

Reply:
doc updated: This field takes effect only when `principal_id` is a `Service Principal` identity.